lib/Parse/ParseExpr.cpp

```diff
@@ -570,7 +570,8 @@ ParserResult<Expr> Parser::parseExprKeyPath() {
       return rootResult;
   }
 
-  if (startsWithSymbol(Tok, '.')) {
+  bool hasLeadingDot = startsWithSymbol(Tok, '.');
```

Comment:
We have this check above (`if(!startsWithSymbol(Tok, '.')`), so maybe we move this check above? 

```c++
bool hasLeadingDot = startsWithSymbol(Tok, '.');
if (!hasLeadingDot) {
  rootResult = ...
  ...
}

if (hasLeadingDot) {
  SyntaxParsingContext ...
  ...
}
```

Reply:
I think it was that way because it’s not expense to check but I guess since we record the result now it’s fair to do it only once :)

Reply:
Actually no, we can't do that because `parseExprPostfix` would move parser forward which means that dot is checked the second time against new `Tok`.